core/arch/instr.h

```diff
@@ -186,7 +186,7 @@ enum {
     /* DR_API EXPORT BEGIN */
     INSTR_DO_NOT_MANGLE = 0x00200000,
     /* DR_API EXPORT END */
-    /* Available: 0x00400000, */
+    INSTR_IS_NOALLOC_STRUCT = 0x00400000,
```

Comment:
Nit: Add a comment here that this flag must only be set by instr_noalloc_init. I'm a bit worried that the "subclass"-like extension will mistakenly be not present if someone misuses this flag. You limited many cases, e.g., when cloning but optionally adding a comment so that everyone remember might help.

Reply:
+1

Reply:
Done.